### PR TITLE
Cleanup

### LINKED ISSUE #106: Cleanup codebase
- [x] Pull from cutting edge NSLS-II stack
- [ ] sidestep warning in mask code `mask.py:185`
~~- [ ] pull `.yml` files from testing from package resources~~ (we don't git clean anymore)
- [x] Move/Remove `ipython_config.py` and `qt_gui.py`
- [ ] Fix flake8 errors

### PR DESCRIPTION
**Pull request by CJ-Wright**
Closes #106

### CODE REVIEW

**Comment by CJ-Wright** [comment]
Cycle for CI?

**Comment by CJ-Wright** [comment]
@chiahaoliu can you take a look at this when you get a chance?

Specifically:
https://github.com/xpdAcq/xpdAn/pull/107/commits/203d577e435f5f7af22d246a7817c201f0cc3ace
https://github.com/xpdAcq/xpdAn/pull/107/commits/1787544e65014053c8f1f1d6fa48f3b60b7126cd
https://github.com/xpdAcq/xpdAn/pull/107/commits/a0c0d38623759f96b1e3ea317900926a7a2430dd

I want to make certain that I'm not breaking anything that I shouldn't have.

Additional code review would be lovely too! I tried to make this a cleanup only doc.

**Comment by chiahaoliu** [comment]
@CJ-Wright all the cleaning looks fine but 1787544e65014053c8f1f1d6fa48f contains ``ENV_VAR`` which is currently used at beamline configuration (related to ``save_tiff``, ``integrate`` and all stuffs). 

If we are ready to move away from the function API permanently , this cleaning should be fine.

**Comment by CJ-Wright** [comment]
You are talking about `XPDAN_SETUP` right? How is that used? Is the beamline always running in setup mode?

**Comment by chiahaoliu** [comment]
Yeah, I was refereeing to ``XPDAN_SETUP``, which is used when generating ``an_glbl`` (for example [here](https://github.com/xpdAcq/xpdAn/blob/master/xpdan/glbl_gen.py#L61)).

The logic was to determine if we are at the beam line, if yes, assign the ``db`` used by ``save_last_tiff`` to the nsls-2 ``db``, otherwise, create a new ``db`` for simulation/testing. We has an edited but uncommitted profile at beam line (which is dirty,I know).

I am abit lost in the header function/analysis ``db`` business, but maybe you can judge if this change will break anything under new workflow?

**Comment by CJ-Wright** [comment]
Hmm, ok. I'd rather not explicitly break the analysis workflow as it exists now. I'm hoping to supplement it with the pipeline based system and then start seceding more functionality to the pipeline. I will add back the file. Efforts on the databroker side may eventually allow us to remove that part of the configuration, where the databroker is searched for via configuration files on the databroker side.

**Comment by CJ-Wright** [comment]
File writer:
- [ ] use map to perform initial transformation of template string
- [ ] use map to perform final transformation of template string
- [ ] push this through a map which saves the data down to disk with human readable name

Image callbacks which need data from raw starts/events (refined parameter vs. temp QOI)
- [ ] push data sets through dummy map which puts datasets together
- [ ] push that to callback sink

**Comment by CJ-Wright** [comment]
Don't forget to write yamls down to disk as well.

**Comment by CJ-Wright** [comment]
Use a emit from any node to combine the background subtracted and not background subtracted branches of the data processing. Also put the calibration filter step after background subtraction.

**Comment by CJ-Wright** [comment]
Have 3 major pipelines all connected to a single raw data source:
1. dark data
1. calibration data
1. regular data

Use the same pipeline chunks for all the pieces where applicable (background subtraction, 
img+calib->pdf, etc.) these will be the same objects in python so the graph will work out properly.
This will also simplify the writing of the code since most of the complex if statement branching/unions will be separated into their own branches.

**Comment by CJ-Wright** [comment]
Closes #108 
Closes #112 
Closes #105 
Closes #34 
Closes #102

**Comment by CJ-Wright** [comment]
TODO:
- [ ] Move dark writing logic into master
- [ ] Replace data_reduction function internals with pipelines
- [ ] Attach writers to pipeline

File: xpdan/tools.py [hunk 1]
```diff
@@ -13,89 +13,202 @@
...
+    a_std: float
+        Acceptable number of standard deviations
+
+    Returns
+    -------
+    list:
+        The positions to be masked
+    """
+    mask_list = []
+    while len(v_list) > 0:
```
Comment (chiahaoliu) [inline xpdan/tools.py]: potentially could use np.apply_over_axis

File: xpdan/tools.py [hunk 2]
```diff
@@ -13,89 +13,202 @@
...
+            break
+        # get the index of the worst pixel
+        worst_idx = np.argmax(norm_v_list)
+        # get the position of the worst pixel
+        worst_p = p_list[worst_idx]
+        # add the worst position to the mask
+        # delete the worst position
+        v_list = np.delete(v_list, worst_idx)
+        p_list = np.delete(p_list, worst_idx)
+        mask_list.append(worst_p)
```
Comment (chiahaoliu) [inline xpdan/tools.py]: instead of appending list, maybe consider creating blanck numpy array first and assigning the index

File: xpdan/tools.py [hunk 3]
```diff
@@ -13,89 +13,202 @@
...
+        tmsk = np.ones(img.shape)
+
+    ring_values = []
+    ring_positions = []
+
+    # TODO: include tmsk into this to speed things up?
+    for i in np.unique(xy):
+        ring_values.append(img.ravel()[xy == i])
+        ring_positions.append(ipos.ravel()[xy == i])
+    with Pool(cpu_count()) as p:
```
Comment (chiahaoliu) [inline xpdan/tools.py]: on the safe side, at least reserve one cpu for main process
eventually, we might want to move to dask

File: xpdan/tools.py [hunk 4]
```diff
@@ -13,89 +13,202 @@
...
+    if tmsk is None:
+        tmsk = np.ones(img.shape)
+
+    ring_values = []
+    ring_positions = []
+
+    # TODO: include tmsk into this to speed things up?
+    for i in np.unique(xy):
+        ring_values.append(img.ravel()[xy == i])
+        ring_positions.append(ipos.ravel()[xy == i])
```
Comment (chiahaoliu) [inline xpdan/tools.py]: can we turn them into ndarray for later broadcasting?

**Pull request merged**